### ISSUE TITLE
[Failed] File: docs.javadsl.ArchiveTest.flowShouldCreateZIPArchive

### ISSUE THREAD

**Issue by ennru** [akka/alpakka#1909]
```
[error] Test docs.javadsl.ArchiveTest.flowShouldCreateZIPArchive failed: java.lang.AssertionError: expected:<ByteString(80, 75, 3, 4, 20, 0, 8, 8, 8, 0, 8, -100, 29, 79, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 19, 0, 0, 0, 97, 107, 107, 97, 95, 102, 117, 108, 108, 95, 99, 111, 108, 111, 114, 46, 115, 118, 103, -19, 85, 73, 110, 27, 49, 16, -4, 74, 99, 114, 110, -118, -5, 18, 88, 2, -30, 32, 71, 61, -62, -98, 40, -110, -31, -79, 108, 88, -54, -56, -48, -21, 83, 69, -59, -122, 125, 10, -112, 91, -128, 0, -94, -72, -11, 82, -84, 46, 114, -82)... and [1078] more> but was:<ByteString(80, 75, 3, 4, 20, 0, 8, 8, 8, 0, 9, -100, 29, 79, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 19, 0, 0, 0, 97, 107, 107, 97, 95, 102, 117, 108, 108, 95, 99, 111, 108, 111, 114, 46, 115, 118, 103, -19, 85, 73, 110, 27, 49, 16, -4, 74, 99, 114, 110, -118, -5, 18, 88, 2, -30, 32, 71, 61, -62, -98, 40, -110, -31, -79, 108, 88, -54, -56, -48, -21, 83, 69, -59, -122, 125, 10, -112, 91, -128, 0, -94, -72, -11, 82, -84, 46, 114, -82)... and [1078] more>, took 0.371 sec

[error]     at docs.javadsl.ArchiveTest.flowShouldCreateZIPArchive(ArchiveTest.java:97)

[error]     ...
```

https://travis-ci.com/akka/alpakka/jobs/229397111#L657

**Comment by michalbogacz** [comment]
It seems that in all error cases, the difference between ZIP files is in metadata of ZIP. I'm not sure why it happens. 
But I can change this test to create ZIP file, then unzip it, and compare the source files with the unzipped.

**Comment by ennru** [comment]
That would be great.

**Comment by ennru** [comment]
Fixed with #1946 (we believe).

**Comment by seglo** [comment]
It's back!

https://travis-ci.com/akka/alpakka/jobs/275839711#L591

**Comment by michalbogacz** [comment]
While creating zip file entry, the current time is taken  https://github.com/openjdk/jdk/blob/master/src/java.base/share/classes/java/util/zip/ZipOutputStream.java#L197
If CI works fast, there is no issue, but if the test will be slow, then this problem occurs. 
The only solution I see right now is to skip bytes representing "File last modification time" in results comparison. 
It does not seems like the cleanest solution, so maybe you have a better idea?

**Comment by ennru** [comment]
You are right, this will always be a problem for CI.
I guess the proper check might be to unzip it and compare only the fixed data instead.

**Comment by ennru** [comment]
Most likely fixed with #2105.
